process/process.go

```diff
@@ -255,6 +264,17 @@ func (p *Process) Output() string {
 	return p.buffer.String()
 }
 
+// Done returns a channel that is closed when the process finishes
+func (p *Process) Done() <-chan struct{} {
+	p.mu.Lock()
+	if p.done == nil {
+		p.done = make(chan struct{})
+	}
+	d := p.done
+	p.mu.Unlock()
+	return d
+}
```

Comment:
This pattern is neat!

Reply:
Stolen from https://golang.org/src/context/context.go, specifically how `context.Done()` works